src/gui/providers/ogr/qgsgeopackageitemguiprovider.cpp

```diff
@@ -344,11 +355,48 @@ void QgsGeoPackageItemGuiProvider::vacuumGeoPackageDbAction( const QString &path
 }
 
 void QgsGeoPackageItemGuiProvider::createDatabase( const QPointer< QgsGeoPackageRootItem > &item )
+{
+  if ( item )
+  {
+    const QgsSettings settings;
+    const QString lastUsedDir = settings.value( QStringLiteral( "UI/lastGeoPackageeDir" ), QDir::homePath() ).toString();
+
+    QString filename = QFileDialog::getSaveFileName( nullptr, tr( "New GeoPackage Database File" ),
+                       lastUsedDir,
+                       tr( "GeoPackage" ) + " (*.gpkg *.GPKG)" );
+    if ( filename.isEmpty() )
+    {
+      return;
+    }
+
+    filename = QgsFileUtils::ensureFileNameHasExtension( filename, QStringList() << QStringLiteral( "gpkg" ) );
+
+    QString errorMsg;
+    if ( QgsProviderMetadata *ogrMetadata = QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "ogr" ) ) )
+    {
+      QString error;
+      if ( ! ogrMetadata->createDatabase( filename, errorMsg ) )
+      {
+        QMessageBox::critical( nullptr, tr( "New GeoPackage Database" ), errorMsg );
```

Comment:
To standardise with similar wording elsewhere:

```suggestion
    QString filename = QFileDialog::getSaveFileName( nullptr, tr( "New GeoPackage" ),
                       lastUsedDir,
                       tr( "GeoPackage" ) + " (*.gpkg *.GPKG)" );
    if ( filename.isEmpty() )
    {
      return;
    }

    filename = QgsFileUtils::ensureFileNameHasExtension( filename, QStringList() << QStringLiteral( "gpkg" ) );

    QString errorMsg;
    if ( QgsProviderMetadata *ogrMetadata = QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "ogr" ) ) )
    {
      QString error;
      if ( ! ogrMetadata->createDatabase( filename, errorMsg ) )
      {
        QMessageBox::critical( nullptr, tr( "New GeoPackage" ), errorMsg );
```